internal/storage/storage_handle.go

```diff
@@ -77,6 +81,19 @@ func NewStorageHandle(ctx context.Context,
 		}),
 		storage.WithPolicy(storage.RetryAlways))
 
-	sh = &storageClient{client: sc}
+	sh = &StorageClient{client: sc}
+	return
+}
+
+func (sh *StorageClient) BucketHandle(bucketName string) (bh *bucketHandle,
+	err error) {
+	storageBucketHandle := sh.client.Bucket(bucketName)
+	attrs, err := storageBucketHandle.Attrs(context.Background())
+
+	if err != nil || attrs == nil {
```

Comment:
what does attrs == nil mean? should we set something to error otherwise, callers will not know what happened

Reply:
We will only check for non-nil err, since attrs == nil signifies some error occurred. Err will be set automatically.

Reply:
are you sure that gcs will set the error incase attrs==nil? Then why do you have a check on attrs==nil?

Reply:
I tried for invalid bucket, it returns attrs as nil and error as "bucket doesn't exist".